core/modules/taxonomy/taxonomy.admin.inc

```diff
@@ -570,9 +586,12 @@ function taxonomy_overview_terms($form, &$form_state, TaxonomyVocabulary $vocabu
   $form['#page_entries'] = $page_entries;
   $form['#back_step'] = $back_step;
   $form['#forward_step'] = $forward_step;
-  $form['#empty_text'] = t('No terms available. <a href="@link">Add term</a>.', array('@link' => url('admin/structure/taxonomy/' . $vocabulary->machine_name . '/add')));
+  $form['#empty_text'] = t('No terms available.');
+  if (taxonomy_vocabulary_access('create', $vocabulary)) {
+    $form['#empty_text'] = t('No terms available. <a href="@link">Add term</a>.', array('@link' => url('admin/structure/taxonomy/' . $vocabulary->machine_name . '/add')));
```

Comment:
```suggestion
    $add_term_link = url('admin/structure/taxonomy/' . $vocabulary->machine_name . '/add');
    $form['#empty_text'] .= ' <a href="@link">' . t('Add term') . '</a>.';
```

Reply:
@klonos I think your suggestion has an error: the new variable doesn't get used.

Reply:
I agree. @klonos if you can let us know what your intent was with this change, maybe we can try something else.